File: ReactNativeApp/ios/tescornappbrownfield/BridgeEvents.swift
import Combine
import Foundation

/// Typed Combine publishers for events emitted by the RN TurboModule layer.
/// The NotificationCenter transport is an internal implementation detail of this framework;
/// consumers subscribe via the Combine interface and never see NotificationCenter directly.
public enum BridgeEvents {
    public static let buttonTapped: AnyPublisher<Void, Never> =
        NotificationCenter.default
            .publisher(for: NSNotification.Name("TescoNativeBridgeButtonTapped"))
            .map { _ in () }
            .eraseToAnyPublisher()
}

File: ReactNativeApp/android/brownfield/src/main/java/com/parser/rnpoc/ReactNativePoC/brownfield/brownfield/BridgeEvents.kt
package com.parser.rnpoc.ReactNativePoC.brownfield.brownfield

import expo.modules.brownfield.BrownfieldMessaging
import kotlinx.coroutines.flow.MutableSharedFlow
import kotlinx.coroutines.flow.SharedFlow
import kotlinx.coroutines.flow.asSharedFlow

/**
 * Typed Kotlin events forwarded from the RN layer via BrownfieldMessaging.
 *
 * Mirrors iOS BridgeEvents.swift (Combine publishers over NotificationCenter).
 * Consumers subscribe to the SharedFlow properties; the BrownfieldMessaging
 * transport is an internal implementation detail of this library.
 *
 * Usage (from consumer app ViewModel):
 *   viewModelScope.launch { BridgeEvents.buttonTapped.collect { /* increment counter */ } }
 */
object BridgeEvents {
  private val _buttonTapped = MutableSharedFlow<Unit>(extraBufferCapacity = 16)
  val buttonTapped: SharedFlow<Unit> = _buttonTapped.asSharedFlow()

  init {
    BrownfieldMessaging.addListener { message ->
      if (message["event"] == "buttonTapped") {
        _buttonTapped.tryEmit(Unit)
      }
    }
  }
}
